### PR TITLE
Fixed everlasting loading of ajax progress bar

### PR DESCRIPTION
**Pull request by nfebe**
Signed-off-by: Fenn-CS <fenn25.fn@gmail.com>
#13130.
Even though I am having problems running the current update #13147. I have tested this with an earlier build and it works just fine.

### CODE REVIEW

**Comment by Fenn-CS** [comment]
Hello @ibennetch ,@nijel, @devenbansod  please someone should review this.

**Comment by nijel** [comment]
Don't worry, we will review this, it's not even submitted for day, just be a bit more patient.

File: js/ajax.js [hunk 1]
```diff
@@ -781,7 +781,7 @@ $(document).on('submit', 'form', AJAX.requestHandler);
  * (e.g: 500 - Internal server error)
  */
 $(document).ajaxError(function (event, request, settings) {
-    if (request.status !== 0) { // Don't handle aborted requests
+     if (!(request.status == 200 && request.status < 300)) {
```
Comment (nijel) [inline js/ajax.js]: This condition seems wrong - the second part is not needed as you've already checked the status == 200.

Also I don't think you should get status 200 without error. Maybe the request.status check is not needed at all? It does come from 16a060e527ed2494f03e1ffe82fe9c47cfab83d0, but there is no good reasoning in the commit message...

File: js/ajax.js [hunk 2]
```diff
@@ -791,7 +791,9 @@ $(document).ajaxError(function (event, request, settings) {
             '<div>' + escapeHtml(errorText) + '</div>' +
             '</div>',
             false
-        );
+            );
         AJAX.active = false;
+    } else if(request.status == 0) {
+        PMA_ajaxShowMessage('An unexpected error happened!');
```
Comment (nijel) [inline js/ajax.js]: What is reason to show different error message in this case?

**Comment by Fenn-CS** [comment]
> This condition seems wrong - the second part is not needed as you've already checked the status == 200.

When server response has been complete/done `(xhr.readyState === 4)`. It then check if the HTTP status is between 200 to 299 (`if (xhr.status == 200 && xhr.status < 300))`, if yes, it means the requested has been successful otherwise it failed `(if !(xhr.status == 200 && xhr.status < 300))`.

> Also I don't think you should get status 200 without error. Maybe the request.status check is not needed at all? It does come from 16a060e, but there is no good reasoning in the commit message.


 I am not checking for status 200 `if(request.status==200)` it the reverse i`f(!(request.status==200))` indicating there was some error.

> What is reason to show different error message in this case?


Well I think its quite unnecessary but status zero errors are special because they happen on the client side making it hard to determine the specific error.

**Comment by nijel** [comment]
What I wanted to say is that following two conditions are equivalent:

```
if (!(request.status == 200 && request.status < 300)) {
```
```
if (!(request.status == 200)) {
```

There is really no need for second condition.

Also you won't get request.status == 200 in the error handler.

Anyway I've spent enough time on this to understand root cause of the problem and I've fixed this myself in https://github.com/phpmyadmin/phpmyadmin/commit/799ac0d4a0d1d6f2cb393c06e7d85e14dd89fa3a

**Pull request closed without merging**